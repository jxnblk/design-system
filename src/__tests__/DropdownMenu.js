import React from 'react'
import renderer from 'react-test-renderer'
import { DropdownMenu } from '..'

describe('DropdownMenu', () => {
  test('renders', () => {
    const json = renderer
      .create(<DropdownMenu>DropdownMenu</DropdownMenu>)
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
  })

  test('renders with "id" prop', () => {
    const json = renderer
      .create(
        <DropdownMenu id="test">Dropdown Menu With "id" Prop</DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
  })

  test('renders with "role" prop', () => {
    const json = renderer
      .create(
        <DropdownMenu role="menu">Dropdown Menu With "role" Prop</DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
  })

  test('renders with "tabIndex" prop', () => {
    const json = renderer
      .create(
        <DropdownMenu tabIndex={0}>
          Dropdown Menu With "tabIndex" Prop
        </DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
  })

  test('renders with "ariaLabelledBy" prop with valid value, e.g. "true" or "false"', () => {
    const json = renderer
      .create(
        <DropdownMenu ariaLabelledBy="test">
          Dropdown Menu With "ariaLabelledBy" Prop
        </DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
  })

  test('renders with "ariaActiveDescendantIndex" prop with invalid value', () => {
    const json = renderer
      .create(
        <DropdownMenu ariaActiveDescendantIndex={5}>
          Dropdown Button With "ariaActiveDescendantIndex" Prop
        </DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
  })

  test('renders with "overflow" prop', () => {
    const json = renderer
      .create(
        <DropdownMenu overflow="hidden">
          Dropdown Button With Cursor
        </DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
    expect(json).toHaveStyleRule('overflow-x', 'hidden')
    expect(json).toHaveStyleRule('overflow-y', 'hidden')
    expect(json).toHaveStyleRule('-webkit-overflow-scrolling', 'hidden')
  })

  test('renders with "height" prop', () => {
    const json = renderer
      .create(
        <DropdownMenu height="88em">Dropdown Button With Cursor</DropdownMenu>
      )
      .toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('-moz-padding-start', '0')
    expect(json).toHaveStyleRule('-webkit-padding-start', '0')
    expect(json).toHaveStyleRule('height', '88em')
  })
})
