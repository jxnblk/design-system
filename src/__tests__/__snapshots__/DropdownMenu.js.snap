// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu renders 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
}

<ul
  aria-activedescendant={null}
  aria-labelledby={undefined}
  className="c0"
  id={undefined}
  role={undefined}
  tabIndex={undefined}
>
  DropdownMenu
</ul>
`;

exports[`DropdownMenu renders with "ariaActiveDescendantIndex" prop with invalid value 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
}

<ul
  aria-activedescendant="mi5"
  aria-labelledby={undefined}
  className="c0"
  id={undefined}
  role={undefined}
  tabIndex={undefined}
>
  Dropdown Button With "ariaActiveDescendantIndex" Prop
</ul>
`;

exports[`DropdownMenu renders with "ariaLabelledBy" prop with valid value, e.g. "true" or "false" 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
}

<ul
  aria-activedescendant={null}
  aria-labelledby="test"
  className="c0"
  id={undefined}
  role={undefined}
  tabIndex={undefined}
>
  Dropdown Menu With "ariaLabelledBy" Prop
</ul>
`;

exports[`DropdownMenu renders with "height" prop 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
  height: 88em;
}

<ul
  aria-activedescendant={null}
  aria-labelledby={undefined}
  className="c0"
  height="88em"
  id={undefined}
  role={undefined}
  tabIndex={undefined}
>
  Dropdown Button With Cursor
</ul>
`;

exports[`DropdownMenu renders with "id" prop 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
}

<ul
  aria-activedescendant={null}
  aria-labelledby={undefined}
  className="c0"
  id="test"
  role={undefined}
  tabIndex={undefined}
>
  Dropdown Menu With "id" Prop
</ul>
`;

exports[`DropdownMenu renders with "overflow" prop 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
  overflow-x: hidden;
  overflow-y: hidden;
  -webkit-overflow-scrolling: hidden;
}

<ul
  aria-activedescendant={null}
  aria-labelledby={undefined}
  className="c0"
  id={undefined}
  overflow="hidden"
  role={undefined}
  tabIndex={undefined}
>
  Dropdown Button With Cursor
</ul>
`;

exports[`DropdownMenu renders with "role" prop 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
}

<ul
  aria-activedescendant={null}
  aria-labelledby={undefined}
  className="c0"
  id={undefined}
  role="menu"
  tabIndex={undefined}
>
  Dropdown Menu With "role" Prop
</ul>
`;

exports[`DropdownMenu renders with "tabIndex" prop 1`] = `
.c0 {
  border: 1px solid #d1d6db;
  border-radius: 2px;
  -moz-padding-start: 0;
  -webkit-padding-start: 0;
}

<ul
  aria-activedescendant={null}
  aria-labelledby={undefined}
  className="c0"
  id={undefined}
  role={undefined}
  tabIndex={0}
>
  Dropdown Menu With "tabIndex" Prop
</ul>
`;
